android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/services/DateService.kt

```diff
@@ -37,4 +37,58 @@ object DateService {
         )
     }
   }
+
+  @JvmOverloads
+  fun addOrSubtractTimeUnitFromCurrentDate(
+    timeUnitCount: Int,
+    operation: String,
+    timeUnit: String = TimeUnit.YEAR.name,
+    dateFormat: String = SDF_YYYY_MM_DD,
+  ): String {
+    when (TimeUnit.valueOf(timeUnit)) {
+      TimeUnit.DAY ->
+        return when (operation) {
```

Comment:
you can extract repeated code into a re-usable variable or function

Reply:
Kindly suggest code changes for clarity.

Reply:
Harmonized